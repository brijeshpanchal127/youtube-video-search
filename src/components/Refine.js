import React, { useState } from 'react';
import { Form } from "react-bootstrap";
import { useDispatch, useSelector } from 'react-redux';
import { refineResults } from '../utils/refine.util';

const Refine = () => {
    const [date, setDate] = useState("");
    const dispatch = useDispatch();
    const results = useSelector(state => state.searchResults)
    const onlyHD = useSelector(state => state.onlyHD)
    const [checked, setChecked] = useState(onlyHD);
    const onChange = (e) => {
    setDate(e.target.value);
    dispatch(refineResults(e.target.value, onlyHD, results));
    }

    const toggleHD = () => {
        dispatch(refineResults(date, !checked, results));
        setChecked(!checked);
    }

    return (
        <Form style={{marginTop:"15px"}}>
            <Form.Group controlId="formBasicEmail">
                <Form.Label style={{ fontWeight: "bold" }}>Refine By Date: (Ex. Jan 01, 2019)</Form.Label>
                <Form.Control type="text" placeholder="Enter date" value={date} onChange={(e) => onChange(e)} />
            </Form.Group>
            <Form.Group controlId="formBasicCheckbox">
                <Form.Check type="checkbox" value={checked} label="Show only HD videos"  onChange={() => toggleHD()} />
            </Form.Group>
        </Form>
    )
}

export default Refine
