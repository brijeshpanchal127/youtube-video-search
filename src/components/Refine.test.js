import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Refine from './Refine';

const mockStore = configureStore([]);

describe('<Refine />', () => {

    let store;
    let component;

    it('should render with initial state from Redux store', () => {
        store = mockStore({
            searchResults: [],
            refinedResults: [],
            refineText: "",
            onlyHD: false
        });
        component = renderer.create(
            <Provider store={store}>
                <Refine />
            </Provider>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

});