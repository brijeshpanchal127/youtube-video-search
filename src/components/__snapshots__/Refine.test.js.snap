// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Refine /> should render with initial state from Redux store 1`] = `
<form
  className=""
  style={
    Object {
      "marginTop": "15px",
    }
  }
>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formBasicEmail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Refine By Date: (Ex. Jan 01, 2019)
    </label>
    <input
      className="form-control"
      id="formBasicEmail"
      onChange={[Function]}
      placeholder="Enter date"
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <div
      className="form-check"
    >
      <input
        className="form-check-input"
        disabled={false}
        id="formBasicCheckbox"
        onChange={[Function]}
        type="checkbox"
        value={false}
      />
      <label
        className="form-check-label"
        htmlFor="formBasicCheckbox"
        title=""
      >
        Show only HD videos
      </label>
    </div>
  </div>
</form>
`;
